### ISSUE TITLE
BUG: read_gguf_vocab returns hidden_size instead of vocab_size — wrong GGUF dimension index

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#353]
## Summary

`read_gguf_vocab()` in `src/model_discovery.cpp` (line 345) returns `dim[1]` of the first GGUF tensor, annotated as `"second dim = vocab"`. But the first tensor (`token_embd.weight` / embedding weight) has GGUF shape `[vocab_size, hidden_size]`, so `dim[0]` is vocab_size and `dim[1]` is hidden_size. The function returns hidden_size, not vocab_size.

## Root Cause

```cpp
// model_discovery.cpp:340-346
for (int j = 0; j < nd; j++) {
    uint64_t dim; fread(&dim, 8, 1, f);
    if (j == 1) { fclose(f); return (int)dim; } // second dim = vocab  ← WRONG, should be j==0
}
```

In GGUF format, tensor dimensions are stored in C order: `dim[0]` = first axis (rows), `dim[1]` = second axis (columns). For `token_embd.weight` shaped `[vocab_size, hidden_size]`, `dim[0]` is vocab_size. This is confirmed by the project's own `gguf_loader.cpp` which reads dimensions the same way and uses `shape[0]` as the first dimension.

## Impact

**Severity: High** — This function is called by `backend_generic.cpp` (line 455) to determine the model's vocab size:

```cpp
// backend_generic.cpp:455
int real_vocab = read_gguf_vocab(path);
if (real_vocab > 0) cfg.vocab = cfg.vocab_size = real_vocab;
load("token_embd.weight", embed, (size_t)real_vocab * H);
```

If hidden_size != vocab_size (almost always the case), this causes:
- `cfg.vocab` = hidden_size instead of actual vocab size
- Embedding buffer allocated with wrong size (e.g., 2048×2048 = 4M floats instead of 262272×2048 = 512M floats)
- Buffer overflow when reading the actual tensor data
- Downstream logits computation uses wrong vocab size, producing garbage outputs or crashes

## Fix

Change `j == 1` to `j == 0`:

```cpp
if (j == 0) { fclose(f); return (int)dim; } // first dim = vocab
```

## Files
- `src/model_discovery.cpp` line ~345
- `src/backend_generic.cpp` line 455 (call site)

**Comment by bong-water-water-bong** [comment]
Fixed in commit 45110185e. Changed dimension index from `j == 1` (which returned hidden_size) to `j == 0` (vocab_size, the first dimension of the embed tensor shaped [vocab_size, hidden_size] in GGUF format).
